client/webpack.config.js

```diff
@@ -0,0 +1,51 @@
+//@ts-check
+
+"use strict";
+
+const path = require("path");
+
+/**
+ * @type {import('webpack').Configuration}
+ */
+const config = {
+    devtool: "source-map",
+    // the entry point of this extension,
+    // 📖 -> https://webpack.js.org/configuration/entry-context/
+    entry: "./src/extension.ts",
+    externals: {
+        // the vscode-module is created on-the-fly and must be excluded.
+        // Add other modules that cannot be webpack'ed, 📖 -> https://webpack.js.org/configuration/externals/
+        vscode: "commonjs vscode",
+        "vscode-languageclient": "vscode-languageclient",
+        "vscode-languageserver-protocol": "vscode-languageserver-protocol",
+    },
+    module: {
+        rules: [{
+            exclude: /node_modules/,
+            test: /\.(ts|js)$/,
+            use: [{
+                loader: "ts-loader",
+                options: {
+                    compilerOptions: {
+                        "module": "es6", // override `tsconfig.json` so that TypeScript emits native JavaScript modules.
+                    },
+                },
+            }],
+        }],
+    },
+    output: {
+        devtoolModuleFilenameTemplate: "../[resource-path]",
+        filename: "extension.js",
+        libraryTarget: "commonjs2",
+        // the bundle is stored in the 'dist' folder (check package.json),
+        // 📖 -> https://webpack.js.org/configuration/output/
+        path: path.resolve(__dirname, "out"),
+    },
+    resolve: {
+        // support reading TypeScript and JavaScript files, 📖 -> https://github.com/TypeStrong/ts-loader
+        extensions: [".ts", ".js"],
+    },
+    // vscode extensions run in a Node.js-context 📖 -> https://webpack.js.org/configuration/node/
+    target: "node",
+};
+module.exports = config;
```

Comment:
I'll assume this was automatically generated somehow.
You know, I'd love to have a few words of context in the PR's description for things like this. I don't want to have to guess which tool you used to generate foo or bar and whatnot xD

And grr, those guys proudly add emojis in the comments but don't even use real arrows. That's nonsense.

Reply:
It is not generated but copy-pasted and then a little bit edited to fit my needs.
I followed this procedure explained by the VSCode team.
https://code.visualstudio.com/api/working-with-extensions/bundling-extension